tmt/steps/provision/testcloud.py

```diff
@@ -435,36 +442,38 @@ def _guess_image_url(self, name):
 
         if not url:
             raise ProvisionError(f"Could not map '{name}' to compose.")
-        return url
+        return cast(str, url)
 
     @staticmethod
-    def _create_template():
+    def _create_template() -> None:
         """ Create libvirt domain template """
         # Write always to ovewrite possible outdated version
         with open(DOMAIN_TEMPLATE_FILE, 'w') as template:
             template.write(DOMAIN_TEMPLATE)
 
-    def wake(self):
+    def wake(self) -> None:
         """ Wake up the guest """
         self.debug(
             f"Waking up testcloud instance '{self.instance_name}'.",
             level=2, shift=0)
         self.prepare_config()
+        assert testcloud is not None
         self._image = testcloud.image.Image(self.image_url)
         self._instance = testcloud.instance.Instance(
             self.instance_name, image=self._image,
             connection=f"qemu:///{self.connection}", desired_arch=self.arch)
 
-    def prepare_ssh_key(self, key_type=None):
+    def prepare_ssh_key(self, key_type: Optional[str] = None) -> None:
         """ Prepare ssh key for authentication """
         # Create ssh key paths
         key_name = "id_{}".format(key_type if key_type is not None else 'rsa')
-        self.key = os.path.join(self.workdir, key_name)
+        assert self.workdir is not None
+        self.key.append(os.path.join(self.workdir, key_name))
```

Comment:
This will be self.key = [.... once #1453 is merged

Reply:
Right, will update it once it gets merged.